polar-core/src/kb.rs

```diff
@@ -664,86 +666,48 @@ impl KnowledgeBase {
         self.resource_blocks.clear();
     }
 
-    fn check_file(&self, src: &str, filename: &str) -> PolarResult<()> {
+    fn check_file(&self, src: &str, filename: &str) -> Result<(), RuntimeError> {
         match (
             self.loaded_content.get(src),
             self.loaded_files.get(filename).is_some(),
         ) {
             (Some(other_file), true) if other_file == filename => {
                 return Err(RuntimeError::FileLoading {
                     msg: format!("File {} has already been loaded.", filename),
-                }
-                .into())
+                })
             }
             (_, true) => {
                 return Err(RuntimeError::FileLoading {
                     msg: format!(
                         "A file with the name {}, but different contents has already been loaded.",
                         filename
                     ),
-                }
-                .into());
+                })
             }
             (Some(other_file), _) => {
                 return Err(RuntimeError::FileLoading {
                     msg: format!(
                         "A file with the same contents as {} named {} has already been loaded.",
                         filename, other_file
                     ),
-                }
-                .into());
+                })
             }
             _ => {}
         }
         Ok(())
     }
 
-    pub fn set_error_context(&self, term: &Term, error: impl Into<PolarError>) -> PolarError {
```

Comment:
👋

Reply:
Ah, sadly we might still need some of this.

For the unhandled partial error. The term that's in the error doesn't itself have source context, since it's a partial constructed in the core. This GetSource visitor was walking the partial looking for the first term that had source info (for example, the original variable that was created).

That context was nice... could we bring it back?

Reply:
https://github.com/osohq/oso/commit/d2644423b47bb562b94fc88edbac7207da2b0ef8